gemma-core/src/test/java/ubic/gemma/persistence/service/common/quantitationtype/QuantitationTypeDaoTest.java

```diff
@@ -44,6 +44,30 @@ public void testLoadValueObjects() {
         assertThat( quantitationTypeDao.loadValueObjects( filters, null ) ).isEmpty();
     }
 
+    @Test
+    public void testFindAllByExperiment() {
+        ArrayDesign ad = createPlatform();
+
+        QuantitationType qt = createQuantitationType( "test" );
+        ExpressionExperiment ee = new ExpressionExperiment();
+
+        BioAssayDimension bad = new BioAssayDimension();
+        sessionFactory.getCurrentSession().persist( bad );
+        RawExpressionDataVector vector = new RawExpressionDataVector();
+        vector.setBioAssayDimension( bad );
+        vector.setDesignElement( ad.getCompositeSequences().iterator().next() );
+        vector.setQuantitationType( qt );
+        vector.setData( new byte[0] );
+        vector.setExpressionExperiment( ee );
+
+        ee.getQuantitationTypes().add( qt );
+        ee.getRawExpressionDataVectors().add( vector );
+
+        sessionFactory.getCurrentSession().persist( ee );
+
+        assertThat( quantitationTypeDao.findByExpressionExperiment( ee ) ).containsExactly( qt );
```

Comment:
It would be good to cover other cases:

 - QTs are not mentioned in `ExpressionExperiment.quantitationTypes`
 - when there is no vector with a given denormalized QT